scalameta/semanticdb-scalac/src/main/scala/scala/meta/internal/HijackAnalyzer.scala

```diff
@@ -2,17 +2,27 @@ package scala.meta.internal
 
 import scala.collection.mutable
 import scala.tools.nsc.{Global => NscGlobal, SubComponent}
+import scala.tools.nsc.typechecker.AnalyzerPlugins
 import scala.tools.nsc.typechecker.SemanticdbAnalyzer
 
 trait HijackAnalyzer extends SemanticdbAnalyzer { self: SemanticdbPlugin =>
 
   def hijackAnalyzer(): Unit = {
     if (!isBatchCompiler) return
 
+    val oldMacroPlugins = {
+      val macroPluginsGetter =
+        classOf[AnalyzerPlugins].getDeclaredMethods.find(_.getName.endsWith("macroPlugins")).get
+      macroPluginsGetter.invoke(global.analyzer).asInstanceOf[List[MacroPlugin]]
+    }
     val newAnalyzer = new { val global: self.global.type = self.global } with SemanticdbAnalyzer
     val analyzerField = classOf[NscGlobal].getDeclaredField("analyzer")
     analyzerField.setAccessible(true)
     analyzerField.set(global, newAnalyzer)
+    oldMacroPlugins.foreach { oldMacroPlugin =>
+      // Restore macroPlugins from old analyzer, see https://github.com/scalameta/scalameta/issues/1135
+      newAnalyzer.addMacroPlugin(oldMacroPlugin.asInstanceOf[newAnalyzer.MacroPlugin])
+    }
```

Comment:
What do you think about copying analyzer plugins, too?

Reply:
Good idea, done.